superset/db_engine_specs/postgres.py

```diff
@@ -145,7 +145,7 @@ def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
         tt = target_type.upper()
         if tt == utils.TemporalType.DATE:
             return f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
-        if tt == utils.TemporalType.TIMESTAMP:
+        if "TIMESTAMP" in tt or "DATETIME" in tt:
```

Comment:
Previously "TIMESTAMP WITH TIMEZONE" was not being picked up, nor was "DATETIME" (`utils.TemporalType` needs a good refactor, too, but will leave that to a follow-up PR)